runelite-client/src/main/java/net/runelite/client/plugins/xptracker/XpTrackerPlugin.java

```diff
@@ -561,4 +566,29 @@ void pauseAllSkills(boolean pause)
 			pauseSkill(skill, pause);
 		}
 	}
+
+	@Subscribe
+	public void onOverlayMenuClicked(OverlayMenuClicked event)
+	{
+		Skill skill = Skill.getSkill(event.getEntry().getTarget());
+		if (skill != null &&
+				event.getEntry().getMenuAction() == MenuAction.RUNELITE_OVERLAY &&
+				event.getEntry().getTarget() != null )
+		{
+			String option = event.getEntry().getOption();
+			if (option.equals(OPTION_RESET))
+			{
+				resetSkillState(skill);
+			}
+			else if (option.equals(OPTION_TOGGLE_TRACKING))
+			{
+				pauseSkill(skill, !xpPauseState.isPaused(skill));
+			}
+			else if (option.equals(OPTION_REMOVE))
+			{
+				xpPanel.toggleCanvasItemText(skill);
```

Comment:
Why is this pausing when it removes?

Reply:
Without the method,when the overlay is removed from the Overlay Menu, the option to "Remove overlay" remains if you right click on the XpInfoBox, meaning the user has to click that option twice to re-add overlay. The method fixes this issue. The method in XpInfoBox that toggles the state was deleted somehow. I'll add it again.